### PR TITLE
BPA Integration

### PR DESCRIPTION
**Pull request by gleibovicipan**
## Status
Ready

## Relates issues
Fixes https://github.com/demisto/etc/issues/19499

## Description
Adds integration for BPA service.
Includes a playbook for running the job and polling for results.

## Required version of Demisto
5.0.0

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [x] Documentation
- [x] Code Review

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] (Integrations/BPA/BPA.yml)
- [ ] (Integrations/BPA/README.md)

### CODE REVIEW

File: Integrations/BPA/BPA.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    notes_checks = feature.get('notes', [])
+    warnings_checks = feature.get('warnings', [])
+    return [transform_check(c, feature_name, category) for c in notes_checks + warnings_checks]
+
+
+def get_results_command(client: Client, args: Dict):
+    task_id = args.get('task_id', '')
+    raw: Dict = client.get_results_request(task_id)
+
+    status = raw.get('status', '')
```
[the remark this answers is not in the document]
Comment (gleibovicipan) [reply Integrations/BPA/BPA.py]: No, will change it to raising error if the response doesn't contain status

File: Integrations/BPA/BPA.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    job_checks: List[Dict] = []
+
+    if status == 'invalid':
+        raise Exception("Job ID not valid or doesn't exist")
+
+    if status == 'complete':
+        results = raw.get('results', {})
+        bpa = results.get('bpa', {})
```
[the remark this answers is not in the document]
Comment (gleibovicipan) [reply Integrations/BPA/BPA.py]: it is the structure of the results object.
it contains other fields like version and timestamp and other information we are not interested in.
The results of the analysis are in the bpa object under results.

**Comment by yuvalbenshalom** [comment]
build still fails. please check.

**Comment by yuvalbenshalom** [comment]
@kirbles19 there are some build issues, but this is generally ready for your review

**Comment by kirbles19** [comment]
@gleibovicipan The integration YML is reviewed. @yaron-libman will review the Playbook YML and update the PR when done.

**Comment by yaron-libman** [comment]
Reviewed the playbook yml.

**Comment by Arsenikr** [comment]
@gleibovicipan @yuvalbenshalom are you handling it?

**Pull request merged**